src/thing_creature.c

```diff
@@ -6078,6 +6078,21 @@ void controlled_continue_looking_excluding_diagonal(struct Thing *creatng, MapSu
     *stl_y = y;
 }
 
+PlayerNumber get_appropriate_player_for_creature(struct Thing *creatng)
+{
+    if ((creatng->alloc_flags & TAlF_IsControlled) != 0)
+    {
+        for (PlayerNumber plyr_idx = 0; plyr_idx < PLAYERS_COUNT; plyr_idx++)
+        {
+            if (is_thing_directly_controlled_by_player(creatng, plyr_idx))
+            {
+                return plyr_idx;
+            }
+        }
+    }
```

Comment:
maybe the if block could be it's own function
so current function would be something in the line of

```
plyr_idx = get_player_controlling_creature(creatng)
if (plyr_idx != INVALID_PLAYER)
    return plyr_idx 
else
  return creatng->owner;
```

would be bit faster to understand

Reply:
or maybe even bit better
```
if (creatng->alloc_flags & TAlF_IsControlled)
    get_player_controlling_creature(creatng)
else
   return creatng->owner;
```